### PR TITLE
frontend/feat #4419 Added deeplink options for customer app

### PR DESCRIPTION
**Pull request by Vignesh-772**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added Deep link options in global payload to make current screen a parent screen and it will exit the app if current screen is parent.
```
deeplinkOptions {
   parent_view :: Maybe Boolean
  , show_title :: Maybe Boolean
}
```
Based on this key we are removing the generic header and handling back press also.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Helpers/Utils.purs [hunk 1]
```diff
@@ -493,6 +496,23 @@ getSearchType _ = do
         Nothing -> "normal_search"
     Nothing -> "normal_search"
 
+getDeepLinkOptions :: LazyCheck -> Maybe DeeplinkOptions
+getDeepLinkOptions _ = do 
+  let mBPayload = unsafePerformEffect $ getGlobalPayload unit
+  case mBPayload of
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Helpers/Utils.purs:502]: can we try `maybe` here

File: Frontend/ui-customer/src/Screens/CustomerUtils/AboutUsScreen/Controller.purs [hunk 2]
```diff
@@ -51,6 +54,9 @@ eval :: Action -> AboutUsScreenState -> Eval Action ScreenOutput AboutUsScreenSt
 
 eval (GenericHeaderActionController (GenericHeaderController.PrefixImgOnClick)) state = continueWithCmd state [do pure BackPressed]
 
-eval BackPressed state = exit $ GoToHomeScreen
+eval BackPressed state = if isParentView FunctionCall then do 
+          _ <- pure $ emitTerminateApp Nothing true
+          continue state
+          else exit $ GoToHomeScreen
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/CustomerUtils/AboutUsScreen/Controller.purs:60]: write with proper indentation and use `void` instead if `_ <-` - 

```purescript
eval BackPressed state = 
    if isParentView FunctionCall 
      then do 
          void $ pure $ emitTerminateApp Nothing true
          continue state
       else 
          exit $ GoToHomeScreen
```

File: Frontend/ui-customer/src/Screens/CustomerUtils/HelpAndSupportScreen/ComponentConfig.purs [hunk 3]
```diff
@@ -145,6 +145,7 @@ genericHeaderConfig state = let
         height = V 25
       , width = V 25
       , imageUrl = fetchImage FF_COMMON_ASSET "ny_ic_chevron_left"
+      , visibility =  if isParentView FunctionCall then GONE else config.prefixImageConfig.visibility
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/CustomerUtils/HelpAndSupportScreen/ComponentConfig.purs:148]: can we move conditions as variables above and use them while generating this

File: Frontend/ui-customer/src/Screens/CustomerUtils/HelpAndSupportScreen/Controller.purs [hunk 4]
```diff
@@ -183,7 +183,10 @@ eval (BackPressed flag ) state = if state.props.isCallConfirmation
   else if state.data.accountStatus == CONFIRM_REQ then continue state{data{accountStatus = ACTIVE}}
   else if state.data.accountStatus == DEL_REQUESTED then updateAndExit (state {data{accountStatus = ACTIVE}} ) $ GoHome
   else if state.props.showDeleteAccountView then continue state{props {showDeleteAccountView = false}}
-  else exit GoBack
+  else if isParentView FunctionCall then do 
+          _ <- pure $ emitTerminateApp Nothing true
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/CustomerUtils/HelpAndSupportScreen/Controller.purs:187]: add `void`

File: Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs [hunk 5]
```diff
@@ -98,7 +99,11 @@ eval (BackPressed backpressState) state = do
     else if state.props.updateProfile then do
       _ <- pure $ hideKeyboardOnNavigation true
       continue state { props { updateProfile = false, genderOptionExpanded = false , expandEnabled = false, isEmailValid = true} }
-        else exit $ GoToHome state
+        else 
+          if isParentView FunctionCall then do 
+          _ <- pure $ emitTerminateApp Nothing true
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs:104]: replace this with `void`

File: Frontend/ui-customer/src/Screens/CustomerUtils/AboutUsScreen/Controller.purs (hunk 2 above)
Comment (witcher-shailesh) [reply Frontend/ui-customer/src/Screens/CustomerUtils/AboutUsScreen/Controller.purs]: Can we have indentation like this?

Comment (Vignesh-772) [reply Frontend/ui-customer/src/Screens/CustomerUtils/AboutUsScreen/Controller.purs]: It is like this only.
Screenshot 2023-11-24 at 9 01 38 AM

Comment (witcher-shailesh) [reply Frontend/ui-customer/src/Screens/CustomerUtils/AboutUsScreen/Controller.purs]: Screenshot 2023-11-24 at 9 04 34 AM
Screenshot 2023-11-24 at 9 04 52 AM
 src="https://github.com/nammayatri/nammayatri/assets/62455315/042eed28-1700-49db-a77b-53da169a4559">

Comment (Vignesh-772) [reply Frontend/ui-customer/src/Screens/CustomerUtils/AboutUsScreen/Controller.purs]: Done.

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**